core/trie/trie.go

```diff
@@ -185,7 +185,9 @@ func (t *Trie) Get(key *felt.Felt) (*felt.Felt, error) {
 		}
 		return nil, err
 	}
-	return value.Value, nil
+	defer nodePool.Put(value)
```

Comment:
It shouldnt be worse than just allocating a new Node whenever we need it.

It could cause our pool to grow for a while, but as the demand decreases Pool will [deallocate](https://pkg.go.dev/sync#Pool) them.

> Any item stored in the Pool may be removed automatically at any time without notification. If the Pool holds the only reference when this happens, the item might be deallocated.